homeassistant/components/lyric/__init__.py

```diff
@@ -77,6 +78,12 @@ async def async_update_data(force_refresh_token: bool = False) -> Lyric:
         try:
             async with asyncio.timeout(60):
                 await lyric.get_locations()
+                for location in lyric.locations:
+                    for device in location.devices:
+                        if device.deviceClass == "Thermostat":
+                            await lyric.get_thermostat_rooms(
+                                location.locationID, device.deviceID
+                            )
```

Comment:
I agree with @t3hk0d3, please use asyncio.gather instead, something like:
```suggestion
                asyncio.gather(
                    lyric.get_thermostat_rooms(
                        location.locationID, device.deviceID
                    )
                    for device in location.devices
                    for location in lyric.locations
                    if device.deviceClass == "Thermostat"
                )
```

Reply:
Updated per suggestion; thanks for the code snippet.  Re-tested locally and it continues to work as expected.